src/app/embed/adapters/feed.adapter.ts

```diff
@@ -10,83 +10,135 @@ const GUID_PREFIX = 's1!';
+      return props;
+    }).catch(err => {
+      FeedAdapter.logError(err);
+      return Observable.of({}); // TODO: really ignore errors?
+    });
+  }
+
+  fetchFeed(feedUrl: string): Observable<string> {
+    let proxied = this.proxyUrl(feedUrl);
+    return this.http.get(proxied).map(res => {
+      if (res.ok && res.text()) {
+        return res.text();
+      } else if (res.ok) {
+        throw new Error(`Got empty response from ${proxied}`);
+      } else {
+        throw new Error(`Got ${res.status} from ${proxied}`);
       }
-      episode = episode || elements[0];
+    });
+  }
 
-      let title = episode.getElementsByTagName('title')[0].textContent;
-      let audioUrl = episode.getElementsByTagName('enclosure')[0].getAttribute('url');
-      let subtitle = doc.getElementsByTagName('title')[0].textContent;
-      let subscribeUrl = doc.getElementsByTagNameNS(atomNamespace, 'link')[0].getAttribute('href');
+  parseFeed(xml: string, episodeGuid: string): AdapterProperties {
+    let parser = new DOMParser();
+    let doc = <XMLDocument> parser.parseFromString(xml, 'application/xml');
+    let props = this.processDoc(doc);
 
-      // Future feeds may not include RP-namespaced data!
-      let feedArtworkUrl  = doc.getElementsByTagNameNS(rpNamespace, 'image')[0].getAttribute('href');
-      let artworkUrl  = episode.getElementsByTagNameNS(rpNamespace, 'image');
+    let episode = this.parseFeedEpisode(doc, episodeGuid);
+    if (episode) {
+      props = this.processEpisode(episode, props);
+    } else {
+      FeedAdapter.logError(`Could not find item with guid ${episodeGuid}`);
+    }
 
-      if (artworkUrl.length === 0) {
-        artworkUrl = feedArtworkUrl;
-      } else {
-        artworkUrl = artworkUrl[0].getAttribute('href');
+    return props;
+  }
+
+  parseFeedEpisode(doc: XMLDocument, episodeGuid: string): Element {
+    let items = doc.querySelectorAll('item');
+    for (let i = 0; i < items.length; i++) {
+      let itemGuid = this.getTagText(items[i], 'guid');
+      if (itemGuid) {
+        if (!this.isEncoded(itemGuid) && this.isEncoded(episodeGuid)) {
+          itemGuid = this.encodeGuid(itemGuid);
+        }
+        if (itemGuid.indexOf(episodeGuid) !== -1) {
+          return items[i];
+        }
       }
+    }
+  }
+
+  processDoc(doc: XMLDocument, props: AdapterProperties = {}): AdapterProperties {
+    props.subtitle = this.getTagText(doc, 'title');
+    props.subscribeUrl = this.getTagAttributeNS(doc, 'atom', 'link', 'href'); // TODO: what if this isn't the first link?
+    props.feedArtworkUrl = this.getTagAttributeNS(doc, 'itunes', 'image', 'href');
+    return props;
+  }
+
+  processEpisode(item: Element, props: AdapterProperties = {}): AdapterProperties {
+    props.title = this.getTagText(item, 'title');
+    props.audioUrl = this.getTagTextNS(item, 'feedburner', 'origEnclosureLink')
+                  || this.getTagAttribute(item, 'enclosure', 'url');
```

Comment:
Hopefully this is what we actually want here.  I stole this from the builder's logic.